pandas/tests/frame/test_missing.py

```diff
@@ -388,6 +388,33 @@ def test_frame_fillna_limit(self):
         expected.values[:3] = np.nan
         tm.assert_frame_equal(result, expected)
 
+    def test_frame_fillna_axis(self):
+        # GH 17399
+        # with limit
+        df = DataFrame(np.random.randn(10, 4))
+        df.iloc[1:4, 1:4] = nan
+        expected = df.copy()
+        expected.iloc[1:4, 1:3] = 0
+        result = df.fillna(value=0, limit=2, axis=1)
+        tm.assert_frame_equal(result, expected)
+        # with no limit
+        expected = df.copy()
+        expected.iloc[1:4, 1:4] = 0
+        result = df.fillna(value=0, axis=1)
+        tm.assert_frame_equal(result, expected)
+        # with method, limit
+        expected = df.copy()
+        for c in lrange(1, 3):
```

Comment:
This can be `expected.iloc[1:4, [1, 2]] = expected.iloc[1:4, 0]`

Reply:
Hmm, that would throw`ValueError: shape mismatch: value array of shape (3,) could not be broadcast to indexing result of shape (2,3)` Error.

Reply:
You could perhaps do 
`expected.iloc[1:4, [1, 2]] = expected.iloc[1:4, 0].values[:, np.newaxis].repeat(2, axis=1)`
opted for loop, since that was more readable.